yt/frontends/chimera/data_structures.py

```diff
@@ -0,0 +1,334 @@
+    def _parse_parameter_file(self):
+        with h5py.File(self.parameter_filename, "r") as f:
+            if "metadata" in f:
+                self.unique_identifier = f["metadata"]["date_string"][
+                    ()
+                ]  # Uses date as unique identifier if available, otherwise uses a default value
+            else:
+                self.unique_identifier = "Chimera"
+            # Reads in simulation time, number of dimensions and shape
+            self.current_time = f["mesh"]["time"][()]
+            self.dimensionality = 3
+            self.domain_dimensions = f["mesh"]["array_dimensions"][()]
+
+            index_filenames = _find_files(self.parameter_filename)
+            if any(
+                "grid_2" in file for file in index_filenames
+            ):  # Checks for Yin-Yang and changes to cartesian
+                self.geometry = "cartesian"
+                self._periodicity = (False, False, False)
+                dre = [
+                    f["mesh"]["x_ef"][(-1)],
+                    f["mesh"]["x_ef"][(-1)],
+                    f["mesh"]["x_ef"][(-1)],
+                ]
+                dle = -1 * np.array(dre)
+                self.domain_dimensions = np.array(
+                    [
+                        2 * f["mesh"]["array_dimensions"][0],
+                        2 * f["mesh"]["array_dimensions"][0],
+                        2 * f["mesh"]["array_dimensions"][0],
+                    ]
+                )
+            else:
+                self.geometry = "spherical"  # Uses default spherical geometry
+                self._periodicity = (False, False, True)
+                dle = [
+                    f["mesh"]["x_ef"][(0)],
+                    f["mesh"]["y_ef"][(0)],
+                    f["mesh"]["z_ef"][(0)],
+                ]
+                if self.domain_dimensions[2] > 2:
+                    dre = [
+                        f["mesh"]["x_ef"][(-3)],
+                        f["mesh"]["y_ef"][(-1)],
+                        f["mesh"]["z_ef"][(13)],
+                    ]
+                else:
+                    dre = [
+                        f["mesh"]["x_ef"][(-1)],
+                        f["mesh"]["y_ef"][(-1)],
+                        f["mesh"]["z_ef"][(-1)],
+                    ]
+            # Sets left and right bounds based on earlier definitions
+            self.domain_right_edge = np.array(dre)
+            self.domain_left_edge = np.array(dle)
+
+        self.cosmological_simulation = 0.0  # Chimera is not a cosmological simulation
+        pass
+
+    @classmethod
+    def _is_valid(self, *args, **kwargs):
+        # This accepts a filename or a set of arguments and returns True or
+        # False depending on if the file is of the type requested.
+        try:
+            fileh = HDF5FileHandler(args[0])
+            if "fluid" in fileh:
+                if "agr_c" in fileh["fluid"].keys():
+                    print("Chimera file identified")  # Numpy bless
```

Comment:
I'm guessing this must be a leftover, but _is_valid methods should not print anything
```suggestion
```

Reply:
It was indeed. Print statement removed in commit 10df6b4.